modules/library/main/src/test/java/org/geotools/feature/visitor/UniqueVisitorTest.java

```diff
@@ -0,0 +1,187 @@
+                    uniqueValuesCountTestType,
+                    new Object[] {10, "C", 30.0, 11, wktParser.read("POINT(5 -5)")},
+                    null),
+            SimpleFeatureBuilder.build(
+                    uniqueValuesCountTestType,
+                    new Object[] {11, "A", 500.0, 12, wktParser.read("POINT(0 0)")},
+                    null),
+            SimpleFeatureBuilder.build(
+                    uniqueValuesCountTestType,
+                    new Object[] {12, "A", 500.0, 12, wktParser.read("POINT(0 0)")},
+                    null),
+            SimpleFeatureBuilder.build(
+                    uniqueValuesCountTestType,
+                    new Object[] {13, "A", 500.0, 12, wktParser.read("POINT(0 0)")},
+                    null),
+        };
+        // creating the feature collection
+        featureCollection = DataUtilities.collection(simpleFeatures);
+    }
+
+    @Test
+    public void testUniqueVisitor() throws IOException {
+        UniqueVisitor uniqueVisitor = new UniqueVisitor("aStringValue");
+        featureCollection.accepts(uniqueVisitor, null);
+        Set result = uniqueVisitor.getResult().toSet();
+        List<String> expected = Arrays.asList("A", "B", "C", "D", "E");
+        assertEquals(expected.size(), result.size());
+        assertTrue(expected.containsAll(result));
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testUniqueVisitor2() throws IOException {
+        UniqueVisitor uniqueVisitor = new UniqueVisitor("aDoubleValue");
+        featureCollection.accepts(uniqueVisitor, null);
+        Set result = uniqueVisitor.getResult().toSet();
+        List expected = Arrays.asList(10d, 20d, 30d, 50d, 60d, 150d, 500d);
+        assertEquals(expected.size(), result.size());
+        assertTrue(result.containsAll(expected));
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testMultipleAttributes() throws IOException {
+        UniqueVisitor uniqueVisitor = new UniqueVisitor("aStringValue", "aDoubleValue");
+        featureCollection.accepts(uniqueVisitor, null);
+        Set result = uniqueVisitor.getResult().toSet();
+
+        assertEquals(10, result.size());
+        List<String> attributes = uniqueVisitor.getAttrNames();
+        assertEquals("aStringValue", attributes.get(0));
+        assertEquals("aDoubleValue", attributes.get(1));
+        LinkedList<Object> shouldContainOnlyOne = new LinkedList<>();
+        shouldContainOnlyOne.add("C");
+        shouldContainOnlyOne.add(20d);
+        LinkedList<Object> shouldContainOnlyOne2 = new LinkedList<>();
+        shouldContainOnlyOne2.add("A");
+        shouldContainOnlyOne2.add(500d);
+        for (Object o : result) {
+            List values = (List) o;
+            assertTrue(values.get(0) instanceof String);
+            assertTrue(values.get(1) instanceof Double);
+        }
+        long count1 = result.stream().filter(o -> o.equals(shouldContainOnlyOne)).count();
```

Comment:
Maybe have a smaller test data set, but do check the result exhaustively like it was done for the single attribute case? I mean, check the result is equal to an expected collection of results.

Reply:
fixed